### PR TITLE
Fix cpp client example

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolong.ran <rxl@apache.org>

Fix cpp client example

Please do not merge this pr before #48  without a fix

### CODE REVIEW

File: cloud/cpp/README.md [hunk 1]
```diff
@@ -20,30 +20,37 @@ $ brew install libpulsar
 
 # Example
 
-In this example, the producer will publish data to the `topic-1` in your Pulsar cluster.
+**Note**
+
+> This example shows how to connect to a Pulsar cluster through OAuth2. To connect to a Pulsar cluster through the token, see the implementation of [ConnectByToken.cc](https://github.com/streamnative/pulsar-examples/blob/master/cloud/cpp/ConnectByToken.cc).
```
Comment (Huanli-Meng) [inline cloud/cpp/README.md:25]:
```suggestion
> This example shows how to connect to a Pulsar cluster through the OAuth2 authentication plugin. To connect to a Pulsar cluster through the token authentication plugin, see the implementation of [ConnectByToken.cc](https://github.com/streamnative/pulsar-examples/blob/master/cloud/cpp/ConnectByToken.cc).
```

File: cloud/cpp/README.md [hunk 2]
```diff
@@ -20,30 +20,37 @@ $ brew install libpulsar
...
+
+In this example, the CPP producer will publish data to the `my-topic` in your Pulsar cluster. The consumer receives the message from the `my-topic` and acknowledges each received message.
```
Comment (Huanli-Meng) [inline cloud/cpp/README.md:27]:
```suggestion
In this example, the CPP producer publishes data to the `my-topic` in your Pulsar cluster. The consumer receives the message from the `my-topic` and acknowledges each received message.
```

File: cloud/cpp/README.md [hunk 3]
```diff
@@ -20,30 +20,37 @@ $ brew install libpulsar
...
 The content of each message payload is  `content`.
-The consumer will receive the message from the `topic-1` and `ack` the receipt of each message received.
 
-> Tips: The following code example uses the OAuth2 connection method. If you want to connect to the Pulsar cluster using Token, please refer to the implementation of **connectByToken.cc**.
+1. Get the Oauth2 options. For details, see [Get Oauth2 options](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-oauth2-authentication-parameters).
```
Comment (Huanli-Meng) [inline cloud/cpp/README.md:30]:
```suggestion
1. Get the Oauth2 authentication parameters. For details, see [Get Oauth2 authentication parameters](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-oauth2-authentication-parameters).
```

File: cloud/cpp/README.md [hunk 4]
```diff
@@ -20,30 +20,37 @@ $ brew install libpulsar
...
 
-1. Run the consumer, and start to receiving the message from `topic-1`:
+2. Get the service URLs. For details, see [Get service URLs](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-pulsar-service-urls).
```
Comment (Huanli-Meng) [inline cloud/cpp/README.md:32]:
```suggestion
2. Get the Pulsar service URLs. For details, see [Get Pulsar service URLs](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-pulsar-service-urls).
```

File: cloud/cpp/README.md [hunk 5]
```diff
@@ -20,30 +20,37 @@ $ brew install libpulsar
...
+Replace the Oauth2 options (`issuer_url`, `audience`, and `private_key`) and the `args[1](serviceURL)` with the values you get from Step 1 and Step 2 respectively.
+
 Output:
 
 ```text
 Received: Message(prod=standalone-1-2, seq=0, publish_time=1596467242114, payload_size=7, msg_id=(71,0,-1,0), props={})  with payload 'content'
 ```
 
-2. Run the producer and publish messages to the `topic-1`:
+4. Run the CPP producer and publish messages to the `my-topic`:
```
Comment (Huanli-Meng) [inline cloud/cpp/README.md:49]:
```suggestion
4. Run the CPP producer and publish messages to the topic `my-topic`.
```

File: cloud/cpp/README.md [hunk 6]
```diff
@@ -20,30 +20,37 @@ $ brew install libpulsar
...
+
+3. Run the CPP consumer to receive messages from `my-topic`:
 
 ```bash
 $ g++ --std=c++11 SampleConsumer.cc -o SampleConsumer -I/usr/local/Cellar/libpulsar/{PULSAR_VERSION}/include -lpulsar -L/usr/local/Cellar/libpulsar/{PULSAR_VERSION}/lib
-$ ./SampleConsumer
+$ ./SampleConsumer pulsar+ssl://cloud.streamnative.dev:6651 '{"issuer_url": "https://cloud.streamnative.dev", "private_key": "/path/to/private/key/file.json", "audience": "urn:sn:pulsar:pulsar-instance-ns:pulsar-instance-name"}'
 ```
 
+Replace the Oauth2 options (`issuer_url`, `audience`, and `private_key`) and the `args[1](serviceURL)` with the values you get from Step 1 and Step 2 respectively.
```
Comment (Huanli-Meng) [inline cloud/cpp/README.md:41]:
```suggestion
Replace the Oauth2 parameters (`issuer_url`, `audience`, and `private_key`) and the `serviceURL` with the values that you get from Step 1 and Step 2 respectively.
```

**Comment by wolfstudy** [comment]
@Huanli-Meng done PTAL again, thanks

File: cloud/cpp/SampleProducer.cc [hunk 7]
```diff
@@ -17,20 +17,18 @@
...
 using namespace pulsar;
 
-int main() {
+int main(int argc, char *argv[]) {
     ClientConfiguration config;
-    std::string params = R"({
-    "issuer_url": "https://dev-kt-aa9ne.us.auth0.com/oauth/token",
-    "private_key": "../../pulsar-broker/src/test/resources/authentication/token/cpp_credentials_file.json",
-    "audience": "https://dev-kt-aa9ne.us.auth0.com/api/v2/"})";
+    std::string params = argv[2];
```
Comment (jiazhai) [inline cloud/cpp/SampleProducer.cc:27]: How about change `params` to `oauthParams`?

Comment (wolfstudy) [reply cloud/cpp/SampleProducer.cc:27]: Cool

**Pull request merged**